File: packages/units/src/weight.ts
export const WEIGHT_UNITS = ["milligram", "gram", "kilogram", "ounce", "pound"] as const;

export type WeightUnit = (typeof WEIGHT_UNITS)[number];

// How many grams equal 1 unit
const GRAM_PER_UNIT: Record<WeightUnit, number> = {
  milligram: 0.001,
  gram: 1,
  kilogram: 1000,
  ounce: 28.349523125,
  pound: 453.59237,
};

export function isWeightUnit(value: string): value is WeightUnit {
  return (WEIGHT_UNITS as readonly string[]).includes(value);
}

export function convertWeight(value: number, from: WeightUnit, to: WeightUnit): number {
  const grams = value * GRAM_PER_UNIT[from];
  return grams / GRAM_PER_UNIT[to];
}

File: src/unit_converter/units/weight.py
from .linear import LinearUnitCategory

WEIGHT_UNITS: tuple[str, ...] = ("milligram", "gram", "kilogram", "ounce", "pound")

# How many grams equal 1 unit.
GRAMS_PER_UNIT: dict[str, float] = {
    "milligram": 0.001,
    "gram": 1,
    "kilogram": 1000,
    "ounce": 28.349523125,
    "pound": 453.59237,
}

_CATEGORY = LinearUnitCategory(WEIGHT_UNITS, GRAMS_PER_UNIT)


def is_weight_unit(value: str) -> bool:
    return _CATEGORY.is_unit(value)


def convert_weight(value: float, from_unit: str, to_unit: str) -> float:
    return _CATEGORY.convert(value, from_unit, to_unit)
